content/news/XXXX-XX-XX-dvs-internals-pt2.md

```diff
@@ -0,0 +1,301 @@
+
+So Serato is using a Linear Feedback Shift Register (LFSR) its control signal.
+Sounds scary?
+Don't worry, it's not as complicated as it may sound.
+
+#### What is an LFSR and how does it work?
+
+A Linear Feedback Shift Register (LFSR) is a shift register that uses a linear feedback function.
+LFSRs are common in modern computing (e.g. for generating pseudo-random numbers), but using them for a vinyl control signal an interesting and unusual application.
+I'll outline the basic principle and then try to explain step-by-step using an example.
+
+For our purposes, it suffices to know that a register is an array that contains a fixed number of bits.
+We are working with a *Shift* Register, which means means that the contents of the register are shifted to the left or right in each step.
+
+![3-bit Shift Register]({static}/images/news/3-bit-shift-register.svg)
+
+If we shift the register's contents to the right, we need a new value that we insert at the leftmost position.
+LFSRs use a (linear) function that takes the current content of the register as input.
+The resulting value is then fed back into the register.
+
+Let's have a look at a step-by-step example.
+Here's a 3-bit LFSR that generates a maximal-length pseudo random bit sequence:
+
+![3-bit maximal-length LFSR]({static}/images/news/3-bit-lfsr.svg)
+
+In each step the following happens:
+
+1. We take the current contents of the register to calculate the feedback bit. In this case, we calculate *x = s<sub>1</sub> + s<sub>0</sub>*.
+2. Then shift the register to the right.
+3. The bit that is "pushed out" of the register (i.e. the rightmost bit *s<sub>0</sub>*) is the output bit
+4. The leftmost (empty) space is filled with the feedback bit (*x*) we calculated.
+
+Let's assume that the initial state of the register is *(1, 0, 0)*.
+
+First, we need to calculate the feedback bit *x* which is defined as the sum of *s<sub>1</sub>* and *s<sub>0</sub>*.
+In this step, both bits have the value `0`, so *x = s<sub>1</sub> + s<sub>0</sub> = 0 + 0 = 0*
+Next, we need to shift everything to the right, and insert *x = 0* on the left.
+The output bit is *0*, because that is the rightmost bit that is "pushed out" of the register.
+
+In the next step, the feedback bit is *x = s<sub>1</sub> + s<sub>0</sub> = 1 + 0 = 0*
+Note that this time *s<sub>1</sub>* has the value *1* because we shifted the register to the right in the previous step and *1* moved from *s<sub>2</sub>* to *s<sub>1</sub>*.
+Now that we calculated *x*, we again shift the whole register to the right and write the feedback bit into the leftmost position.
+The output bit is *0*.
+
+We can now continue this a few more times.
+In step 5, the feedback bit is *x = s<sub>1</sub> + s<sub>1</sub> = 1 + 1*.
+Usually, the result of that calculation would be *2*, but in this case it's *0*.
+The reason for that is that we're working with bits (which are either *0* or *1*), so *2* is not a valid value.
```

Comment:
```suggestion
The reason for that is the value range overflow. Since we work with bits whose values can be either *0* or *1*, the invalid value *2* is represented as *0*.```

Reply:
Tbh I like the original phrasing better, I think it's easier to understand for less technically inclined people.